lib/opencensus/trace/span_builder.rb

```diff
@@ -123,17 +146,149 @@ def finish!
         self
       end
 
+      ##
+      # Add an attribute to this span.
+      #
+      # Attributes are key-value pairs representing properties of this span.
+      # You could, for example, add an attribute indicating the URL for the
+      # request being handled, the user-agent, the database query being run,
+      # the ID of the logged-in user, or any other relevant information.
+      #
+      # Keys must be strings.
+      # Values may be String, TruncatableString, Integer, or Boolean.
+      # The valid integer range is 64-bit signed `(-2^63..2^63-1)`.
+      #
+      # @param [String, Symbol] key
+      # @param [String, TruncatableString, Integer, Boolean] value
+      #
+      def put_attribute key, value
+        @attributes[key.to_s] = value
+      end
+
+      ##
+      # Add an event annotation with a timestamp.
+      #
+      # @param [String] description Description of the event
+      # @param [String] attributes Key-value pairs providing additional
+      #     properties of the event. Keys must be strings, and values are
+      #     restricted to the same types as attributes (see #put_attribute).
+      # @param [Time] time Timestamp of the event (optional, defaults to the
+      #     current time)
+      #
+      def put_annotation description, attributes = {}, time: nil
```

Comment:
Should attributes be a named parameter (like the other optional values) rather than positional?

Reply:
I rather preferred to keep this one positional since it's already a (possibly empty) hash of optional attributes. Making it a named parameter makes for clumsy calling conventions, especially because then you almost have to do the same for put_link.